src/Components/test/testassets/BasicTestApp/PreserveStateService.cs

```diff
@@ -11,38 +11,45 @@ public class PreserveStateService : IDisposable
     {
         private readonly ComponentApplicationState _componentApplicationState;
 
+        private ServiceState _state = new();
+
         public PreserveStateService(ComponentApplicationState componentApplicationState)
         {
             _componentApplicationState = componentApplicationState;
             _componentApplicationState.OnPersisting += PersistState;
             TryRestoreState();
         }
 
-        public Guid Guid { get; private set; }
+        public Guid Guid => _state.TheState;
 
         private void TryRestoreState()
         {
-            if (_componentApplicationState.TryRedeemFromJson<Guid>("Service", out var guid))
+            if (_componentApplicationState.TryRedeemFromJson<ServiceState>("Service", out var state))
```

Comment:
This this because S.T.J. doesn't deal with `Guid` as the top-level type being (de)serialized?

Reply:
No, it's because the serialization options didn't exercise serializing properties, so want to make sure the test covers roundtripping objects.

Reply:
That makes sense. Thanks!